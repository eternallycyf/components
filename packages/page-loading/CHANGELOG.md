# Changelog

### [Version&nbsp;1.3.1](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.3.0...@ims-view/page-loading@1.3.1)

<sup>Released on **2024-04-28**</sup>

#### 🐛 修复

- Keep react version.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- Keep react version ([fee7e8d](https://github.com/eternallycyf/components/commit/fee7e8d))

</details>

## [Version&nbsp;1.3.0](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.2.0...@ims-view/page-loading@1.3.0)

<sup>Released on **2024-02-09**</sup>

#### ✨ 新特性

- **@ims-view/calendar**: Add MiniCalendar.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/calendar**: Add MiniCalendar ([6be604f](https://github.com/eternallycyf/components/commit/6be604f))

</details>

## [Version&nbsp;1.2.0](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.1.3...@ims-view/page-loading@1.2.0)

<sup>Released on **2024-01-12**</sup>

#### ✨ 新特性

- Test publish:.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Test publish: ([2a6716c](https://github.com/eternallycyf/components/commit/2a6716c))

</details>

### [Version&nbsp;1.1.3](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.1.2...@ims-view/page-loading@1.1.3)

<sup>Released on **2023-12-27**</sup>

#### 🐛 修复

- **@ims-view/page-loading**: Style use css.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **@ims-view/page-loading**: Style use css ([3ce234a](https://github.com/eternallycyf/components/commit/3ce234a))

</details>

### [Version&nbsp;1.1.2](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.1.1...@ims-view/page-loading@1.1.2)

<sup>Released on **2023-12-27**</sup>

#### 🐛 修复

- **@ims-view/page-loading**: Style use webkit prefix.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **@ims-view/page-loading**: Style use webkit prefix ([66cec9c](https://github.com/eternallycyf/components/commit/66cec9c))

</details>

### [Version&nbsp;1.1.1](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.1.0...@ims-view/page-loading@1.1.1)

<sup>Released on **2023-12-27**</sup>

#### 🐛 修复

- **@ims-view/page-loading**: Style use webkit prefix.

#### 🔖 杂项

- **自动发布**: @ims-view/page-loading@1.1.0 [skip ci].

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- **@ims-view/page-loading**: Style use webkit prefix ([71f5ea1](https://github.com/eternallycyf/components/commit/71f5ea1))

#### Chores

- **自动发布**: @ims-view/page-loading@1.1.0 [skip ci] ([eee7e5f](https://github.com/eternallycyf/components/commit/eee7e5f))

</details>

## [Version&nbsp;1.1.0](https://github.com/eternallycyf/components/compare/@ims-view/page-loading@1.0.0...@ims-view/page-loading@1.1.0)

<sup>Released on **2023-12-26**</sup>

#### ✨ 新特性

- **@ims-view/page-loading**: Add PageLoading.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/page-loading**: Add PageLoading ([de0feaa](https://github.com/eternallycyf/components/commit/de0feaa))

</details>

## Version&nbsp;1.0.0

<sup>Released on **2023-12-26**</sup>

#### ✨ 新特性

- **@ims/page-loading**: Add page-loading.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims/page-loading**: Add page-loading ([233d68e](https://github.com/eternallycyf/components/commit/233d68e))

</details>
