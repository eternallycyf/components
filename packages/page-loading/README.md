# @ims-view/page-loading

[![NPM version][version-image]][version-url] [![NPM downloads][download-image]][download-url]

## License

[MIT](../../LICENSE) Â® eternallycyf

<!-- npm url -->

[version-image]: http://img.shields.io/npm/v/@ims-view/page-loading.svg?color=deepgreen&label=latest
[version-url]: http://npmjs.org/package/@ims-view/page-loading
[download-image]: https://img.shields.io/npm/dm/@ims-view/page-loading.svg
[download-url]: https://npmjs.org/package/@ims-view/page-loading

<!-- repo url -->

- https://github.com/eternallycyf/components
