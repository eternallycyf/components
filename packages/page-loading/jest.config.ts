import { Config } from '@umijs/test';
import base from '../../jest.config.base';

const packageName = '@ims-view/page-loading';

const root = '<rootDir>/packages/page-loading';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
};

export default config;
