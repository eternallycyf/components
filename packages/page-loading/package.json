{
  "name": "@ims-view/page-loading",
  "version": "1.4.0",
  "main": "lib/index.js",
  "unpkg": "dist/index.js",
  "module": "es/index.js",
  "types": "es/index.d.ts",
  "files": [
    "lib",
    "es",
    "dist"
  ],
  "scripts": {
    "build": "father build",
    "clean": "rm -rf es lib dist build coverage .eslintcache",
    "cov": "jest --coverage",
    "doctor": "father doctor",
    "prepublishOnly": "npm run doctor && npm run build",
    "start": "father dev",
    "test": "jest"
  },
  "dependencies": {
    "@babel/runtime": "^7.23.1",
    "multi-nprogress": "^0.3.5",
    "insert-css": "^2.0.0",
    "react-error-overlay": "^6.0.9"
  },
  "devDependencies": {
    "@types/nprogress": "^0.2.3",
    "@types/insert-css": "^2.0.0"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  }
}
