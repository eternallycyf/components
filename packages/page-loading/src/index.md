---
title: 说明
order: 0
nav:
  title: 基础组件
  order: 4
apiHeader: false
---

## 说明

一些简单的基础组件, 拿来即用。

##
