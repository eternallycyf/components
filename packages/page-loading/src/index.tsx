export { default as PageLoading } from './page-loading';
export { default as Square } from './page-loading/preloader';
