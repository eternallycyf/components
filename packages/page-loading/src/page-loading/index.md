---
title: PageLoading
order: 2
group: 信息展示
apiHeader: false
---

# PageLoading

[![NPM version][version-image]][version-url] [![NPM downloads][download-image]][download-url]

<!-- npm url -->

[version-image]: http://img.shields.io/npm/v/@ims-view/page-loading.svg?color=deepgreen&label=latest
[version-url]: http://npmjs.org/package/@ims-view/page-loading
[download-image]: https://img.shields.io/npm/dm/@ims-view/page-loading.svg
[download-url]: https://npmjs.org/package/@ims-view/page-loading

页面加载组件

## 特性

- 支持全屏；
- 吸顶加载进度条；
- 自定义主色；
- 自定义加载图形；

## 使用案例

### 默认用法

<code src="./demos/Square" ></code>

<code src="./demos/Basic" ></code>

### 不带进度条

<code src="./demos/WithoutProgress" ></code>

### 自定义颜色

<code src="./demos/CustomColor" ></code>

### 全屏加载

<code src="./demos/Fullscreen" ></code>

### 加载状态控制

<code src="./demos/LoadingState" ></code>

### 自定义加载图形

<code src="./demos/CustomLoader" ></code>
