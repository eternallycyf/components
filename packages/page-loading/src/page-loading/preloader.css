@-webkit-keyframes pulse {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
.component-preloader-square-container {
  display: flex;
  align-items: center;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 36px;
  height: 36px;
}

.component-preloader-square-item {
  width: 35%;
  height: 35%;
  animation: 0.75s ease-in infinite alternate pulse;
}

.component-preloader-square-animation-delay {
  -webkit-animation-delay: 0.75s;
  animation-delay: 0.75s;
}
