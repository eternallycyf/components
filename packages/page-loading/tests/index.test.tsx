import { render } from '@testing-library/react';

import { PageLoading } from '@ims-view/page-loading';
import React from 'react';

test('PageLoading', () => {
  const { container } = render(<PageLoading />);

  expect(container).toMatchSnapshot();
});
