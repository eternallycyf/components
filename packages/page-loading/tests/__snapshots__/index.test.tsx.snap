// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading 1`] = `
<div>
  <div
    class="component-page-loading-container undefined nprogress-custom-parent"
    id="component-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="components-page-loading-loader"
    >
      <div
        class="component-preloader-square-container"
      >
        <div
          class="component-preloader-square-item"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="component-preloader-square-item component-preloader-square-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="component-preloader-square-item component-preloader-square-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="component-preloader-square-item"
          style="background: rgb(24, 144, 255);"
        />
      </div>
    </div>
    <div
      class="nprogress"
      id="nprogress"
    >
      <div
        class="bar"
        role="bar"
        style="transition: all 200ms linear; transform: translate3d(-100%,0,0); margin-left: -82%;"
      >
        <div
          class="peg"
        />
      </div>
    </div>
  </div>
</div>
`;
